os/arch/arm/src/amebasmart/amebasmart_spi.c

```diff
@@ -1002,8 +1135,66 @@ static void amebasmart_spi_exchange_nodma(FAR struct spi_dev_s *dev,
 			}
 		}
 	}
+#else
+/*************************************************************************
+ * Name: spi_exchange (with DMA)
+ *
+ * Description:
+ *   Exchange a block of data on SPI using DMA
+ *
+ * Input Parameters:
+ *   dev      - Device-specific state data
+ *   txbuffer - A pointer to the buffer of data to be sent
+ *   rxbuffer - A pointer to a buffer in which to receive data
+ *   nwords   - the length of data to be exchanged in units of words.
+ *              The wordsize is determined by the number of bits-per-word
+ *              selected for the SPI interface.  If nbits <= 8, the data is
+ *              packed into uint8_t's; if nbits >8, the data is packed into uint16_t's
+ *
+ * Returned Value:
+ *   None
+ *
+ ************************************************************************************/
+	uint8_t *rxbuff = NULL;
+	rxbuff = (uint8_t *)rtw_zmalloc(SPI_DMA_MAX_BUFFER_SIZE);
+	if(rxbuff == NULL) {
+		return;
+	}
+	uint8_t *txbuff =  NULL;
+	txbuff = (uint8_t *)rtw_zmalloc(SPI_DMA_MAX_BUFFER_SIZE);
+	if(txbuff == NULL) {
+		return;
+	}
+	int mode_16bit = 1;
+	spi_hookmastercallback(priv);
+	if (amebasmart_spi_9to16bitmode(priv)) {
+		/* 16-bit mode */
+		mode_16bit++;
+	}
+
+	if (txbuffer && rxbuffer) {
+		if (nwords * mode_16bit > 65535) PANIC();
```

Comment:
Why do you call the `PANIC`? It should be never happened?

Reply:
In case there is a large SPI data >65535 (maximum size of DMA block) passed in from application.

Reply:
Just returning error is not enough? PANIC is necessary?

Reply:
I have changed it to just DEBUGASSERT, and malloc rx/tx only if it is required.

Reply:
ASSERT or PANIC causes a device reset. Please let me know why you need it. When it gets bigger than 65535, what happened? Does it cause a device non-recoverable error?